### PR TITLE
Surface the LastErrorValue in json output

### LINKED ISSUE #16: Surface the contents of the LastErrorValue/GetLastError() in the output
In bug [1692309](https://bugzilla.mozilla.org/show_bug.cgi?id=1692309) we added functionality to surface the contents of the last error caused by a system call on Windows. We should include the [relevant patch](https://searchfox.org/mozilla-central/rev/f07a609a76136ef779c65185165ff5ac513cc172/toolkit/crashreporter/breakpad-patches/16-get-last-error.patch) in the stackwalker build and modify the output to include the contents of the last error when the value is non-zero.

Currently the stackwalker will spit out an array of threads with each element containing a `frame_count` attribute and a `frames` array holding the thread frames like so:
```
{
  // ...
  "threads": [
    {
      "frame_count": 10,
      "frames": [...]
    },
    // ...
  ]
  // ...
}
```
We could add the last error value in each entry of the `threads` array when it is non-zero. Printing it out as hex is probably the best way as that's how errors are described in Microsoft's documentation. So a thread entry might look like this:
```
{
  "frame_count": 10,
  "frames": [...],
  "last_error_value": "0x2", // This is would be a thread that encountered a ERROR_FILE_NOT_FOUND error
}
```

### PR DESCRIPTION
**Pull request by Gankra**
I thiiiink this should work but I still need to figure out how to load these changes into the docker image for testing

fixes #16

### CODE REVIEW

**Comment by willkg** [comment]
You can build a new Docker image with `make build`. Does that help at all?

**Comment by Gankra** [comment]
huh, I am completely baffled at why this hunk is being rejected: 

```
$ cat breakpad/src/src/processor/call_stack.cc.rej
--- src/processor/call_stack.cc
+++ src/processor/call_stack.cc
@@ -44,11 +44,12 @@ CallStack::~CallStack() {

 void CallStack::Clear() {
   for (vector<StackFrame *>::const_iterator iterator = frames_.begin();
        iterator != frames_.end();
        ++iterator) {
     delete *iterator;
   }
   tid_ = 0;
+  last_error_ = 0;
 }

 }  // namespace google_breakpad
```

File: minidump-stackwalk/stackwalker.cc [hunk 1]
```diff
@@ -803,6 +803,7 @@ static void ConvertProcessStateToJSON(const ProcessState& process_state,
     }
     thread["frames"] = stack;
     thread["frame_count"] = stack.size();
+    thread["last_error_value"] = ToHex(raw_stack.last_error());
```
Comment (gabrielesvelto) [inline minidump-stackwalk/stackwalker.cc]: print this out only if `raw_stack.last_error()` is non-zero

File: breakpad-patches/16-get-last-error.patch [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+   // MDRawThread structure.  Returns false if the thread ID cannot be
+   // determined.
+   virtual bool GetThreadID(uint32_t *thread_id) const;
+diff --git a/src/processor/call_stack.cc b/src/processor/call_stack.cc
+--- a/src/processor/call_stack.cc
++++ b/src/processor/call_stack.cc
+@@ -44,11 +44,12 @@ CallStack::~CallStack() {
+ 
+ void CallStack::Clear() {
+   for (vector<StackFrame *>::const_iterator iterator = frames_.begin();
```
Comment (gabrielesvelto) [inline breakpad-patches/16-get-last-error.patch]: This line reads:
```
   for (vector<StackFrame*>::const_iterator iterator = frames_.begin();
```
In Breakpad sources, I think this is why the patch doesn't apply.

**Review by gabrielesvelto — CHANGES_REQUESTED** [review]
Just a couple of changes are needed, the rest is fine.

**Comment by Gankra** [comment]
comments addressed / patch fixed

**Review by gabrielesvelto — APPROVED** [review]

**Pull request merged**